### PR TITLE
Campaign Dashboard: Fix incompatibility with CiviCRM 5.67+

### LINKED ISSUE #114: Campaign Dashboard: replace calls to dead core BAO methods - incompatibility with CiviCRM 5.67+
```
 ------ ----------------------------------------------------------------------------------- 
  Line   CRM/CampaignManager/CampaignTree/Page/Dashboard.php                                
 ------ ----------------------------------------------------------------------------------- 
  104    Call to an undefined static method CRM_Campaign_BAO_Campaign::getCampaignCount().  
  133    Call to an undefined static method CRM_Campaign_BAO_Survey::getSurveyCount().      
  162    Call to an undefined static method CRM_Campaign_BAO_Petition::getPetitionCount().  
 ------ ----------------------------------------------------------------------------------- 
```

Maybe someone can estimate the implications?

_internal ref: 22950_

### PR DESCRIPTION
**Pull request by CharlotteJana**
Fixes #114.

_systopia-reference: 22950_

### CODE REVIEW

**Review by jensschuppe — COMMENTED** [review]
Looks good! I just `use`d the APIv4 entity classes.

File: CRM/CampaignManager/CampaignTree/Page/Dashboard.php [hunk 1]
```diff
@@ -130,7 +137,10 @@ public function browseSurvey() {
 
     $this->assign('addSurveyUrl', CRM_Utils_System::url('civicrm/survey/add', 'reset=1&action=add'));
 
-    $surveyCount = CRM_Campaign_BAO_Survey::getSurveyCount();
+    $surveyCount = Survey::get(FALSE)
+      ->selectRowCount()
+      ->execute()
+      ->countMatched();
```
Comment (colemanw) [inline CRM/CampaignManager/CampaignTree/Page/Dashboard.php:143]: @jensschuppe I see that this does [exactly what the old function did](https://github.com/civicrm/civicrm-core/blob/5.66/CRM/Campaign/BAO/Survey.php#L197-L199), so it's technically the right fix... but I also thought I'd point out that this has a weird asymmetry to it. The substitute `getPetitionCount` code below fetches all surveys of type "Petition" but this code just fetches all surveys (and does not exclude "Petition") so the totals won't add up.

Comment (jensschuppe) [reply CRM/CampaignManager/CampaignTree/Page/Dashboard.php:143]: Thanks for pointing that out, we stumbled upon that as well, but that's the same thing Core does/did for deciding whether to continue with whatever needs to be done for each campaign dashboard tab. This code seems to have been copied from Core when it was built a long time ago.

Also, as this extension is to be phased out in favor of https://github.com/ixiam/campaignmanager, so we chose to just do the *technically correct fix*.

Comment (colemanw) [reply CRM/CampaignManager/CampaignTree/Page/Dashboard.php:143]: Sounds good!

**Pull request merged**